codeqlAlerts.js

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env node
+"use strict";
+
+require("dotenv").config();
+
+const _ = require("lodash");
+const moment = require("moment");
+const owner = "Sparkpost";
+const Promise = require("bluebird");
+const { Octokit } = require("@octokit/rest");
+const octokit = new Octokit({
+  auth: process.env.GITHUB_TOKEN,
+  // Set GitHub Auth Token in environment variable
+});
+
+const [, , ...args] = process.argv;
+
+//  .number, .created_at, .url, .html_url, .state, .dismissed_by.login, .dismissed_at, .dismissed_reason, .rule.id, .rule.severity, .rule.description, .tool.name, .most_recent_instance.classifications[]]
+async function getCodeAlerts(repos) {
+  return Promise.map(repos, async ({ name, org }) => {
+    const sortedAlerts = {};
+    const summary = {};
+
+    return await octokit
+      .paginate(octokit.codeScanning.listAlertsForRepo, {
+        owner: org,
+        repo: name,
+      })
+      .then(async (alerts) => {
+        const filteredAlerts = filterCodeAlerts(alerts);
+
+        filteredAlerts.forEach((alert) => {
+          var rule = alert.rule.description;
+          if (!sortedAlerts[rule]) {
+            sortedAlerts[rule] = { count: 1 };
+            sortedAlerts[rule]["createdAt"] = alert.created_at;
+            sortedAlerts[rule]["severity"] = alert.rule.severity;
+          } else {
+            sortedAlerts[rule]["count"] += 1;
+          }
+          if (alert.rule.severity === "error") {
+            summary.error = (summary.error || 0) + 1;
+          }
+          if (alert.rule.severity === "warning") {
+            summary.warning = (summary.warning || 0) + 1;
+          }
+        });
+        return sortedAlerts;
+      })
+      .then(async (sortedAlerts) => {
+        const blocks = Object.keys(sortedAlerts).map((alert) =>
+          buildBlocks("code", alert, sortedAlerts[alert])
+        );
+        return { repo: name, summary, blocks };
+      });
+  }).catch((error) => {
+    console.error(
+      `Failed for ${org}/${name}\n${error.message}\n${error.documentation_url}`
+    );
+  });
+}
+
+function filterCodeAlerts(alerts) {
+  return alerts.filter((alert) => {
+    const remediationDue = moment(alert.created_at).add(14, "days");
+    if (
+      alert.rule.severity !== "note" &&
+      alert.most_recent_instance.classifications !== "test" &&
+      remediationDue < moment()
+    ) {
+      return alert;
```

Comment:
`.filter` will include only items that evaluate as true.. so really, I think you just need:
```
return alerts.filter((alert) => (
  alert.rule.severity !== "node" &&
  alert.most_recent_instance.classifications !== "test" &&
  (moment(alert.created_at).add(14, "days")) < moment()
));
```